pkg/plugins/validation.go

```diff
@@ -0,0 +1,121 @@
+package plugins
+
+import (
+	"embed"
+	"path"
+
+	"github.com/ghodss/yaml"
+	"github.com/xeipuuv/gojsonschema"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/kumahq/kuma/pkg/core/validators"
+)
+
+//go:embed policies/*/api/v1alpha1/schema.yaml
```

Comment:
how about we make it more local? Instead of doing this wildcard (+workaround) and mapping resourceToSchema, we can load specific schema in plugin package and pass embed.FS (or file etc.) to `ValidateResourceSchema`.

Reply:
and then:
1. Would `ValidateResourceSchema` be called from a plugin?
2. Do you want me to create a sort of registry here where a plugin would register schema for it's type and the call site remains the same?

For 1:
- pros
  - no need to have this weird [null check](https://github.com/kumahq/kuma/pull/5107/files#diff-e8050d06f1e67dc5ce3a6f7d395cfe3c864a57871075d56d3dfa194d91dfd85cR36) 
- cons
  - multiple call sites for schema validation

For 2 (the inverse for 1):
- pros
  - one call site for schema validation
- cons
  - [null check](https://github.com/kumahq/kuma/pull/5107/files#diff-e8050d06f1e67dc5ce3a6f7d395cfe3c864a57871075d56d3dfa194d91dfd85cR36) is still there

Reply:
talked offline, went for option 1